### ISSUE TITLE
[fix] labels must be correctly programmatically associated in file input examples

### ISSUE THREAD

**Issue by shkeating** [trussworks/react-uswds#2609]
**ReactUSWDS Version: 5.4.0 & USWDS Version: 3.6.0**

# Describe the bug

In our storybook example of the File Input component with an error, we do not correctly associate its label.

The following examples for this component do not have a _programmatically associated_ label. There is a `label` present, however, it has a `for=` attribute with the id of another example in the file.

- [file input with error](https://trussworks.github.io/react-uswds/?path=/story/components-file-input--with-error)

# To Reproduce

1. Go to [file input with error](https://trussworks.github.io/react-uswds/?path=/story/components-file-input--with-error)
2. run an automated accessibility check
3. see test failure for no label for the With Error example

# Expected behavior

The label's `for` attribute must match the id of the input it is associated with. In this case, we want our label to have `for="file-input-multiple"` to be `for="file-input-error"` instead. this was likely overlooked int he initial implementation of this page.

# Additional context

- [Labelling Controls - W3C](https://www.w3.org/WAI/tutorials/forms/labels/)
- file(s) to be edited: `src/components/forms/FileInput/FileInput.stories.tsx` see line 88 for the error example.

# Screenshots

<img width="1313" alt="image" src="https://github.com/trussworks/react-uswds/assets/59394696/42c00244-238f-47e1-88a6-57114c2ead84">

<img width="1309" alt="image" src="https://github.com/trussworks/react-uswds/assets/59394696/4266456e-ebbd-43a9-8e2b-d871ed598cba">
